cmd/snap/cmd_run.go

```diff
@@ -1492,7 +1497,7 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 		env["XAUTHORITY"] = xauthPath
 	}
 
-	if len(krb5ccnamePath) > 0 {
+	if !needsClassic && len(krb5ccnamePath) > 0 {
```

Comment:
len(krb5ccnamePath) == 0 for classic already, maybe instead of adding a new condition it's enough to leave a comment, e.g. 
```
// we have a new location for the ticket, update the environment variable
```

Reply:
The suggestion also includes removing the `!needsClassic` condition.

Reply:
I missed that.